schematic/store/synapse.py

```diff
@@ -706,11 +706,21 @@ def getFilesInStorageDataset(
         """
         file_list = []
 
-        # Get path to dataset folder from fileview to avoid building a new fileview and walking to determine folders and files within
+        # Get path to dataset folder by using childern to avoid cases where the dataset is the scope of the view
         child_path = self.storageFileviewTable.loc[
             self.storageFileviewTable["parentId"] == datasetId, "path"
-        ][0]
-        parent = child_path.split("/")[0]
+        ]
+        if child_path.empty:
```

Comment:
You'll want to add a test case for this to ensure this error is raised (no need for integration test, just a mocked test)